### PR TITLE
[RW-7856][risk=no] Updated with logic to use OMOP 5.3.1 schema changes

### PR DESCRIPTION
**Pull request by chenchals**
Description:

- Updated for OMOP v5.3.1 - May 2022 changes that affect tests and CDR indices build

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 1]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+  "INSERT INTO \`$BQ_PROJECT.$BQ_DATASET.ds_condition_occurrence\`
+      (PERSON_ID, CONDITION_CONCEPT_ID, STANDARD_CONCEPT_NAME, STANDARD_CONCEPT_CODE, STANDARD_VOCABULARY,
+      CONDITION_START_DATETIME, CONDITION_END_DATETIME, CONDITION_TYPE_CONCEPT_ID, CONDITION_TYPE_CONCEPT_NAME,
+      STOP_REASON, VISIT_OCCURRENCE_ID, VISIT_OCCURRENCE_CONCEPT_NAME, VISIT_DETAIL_ID, CONDITION_SOURCE_VALUE, CONDITION_SOURCE_CONCEPT_ID,
+      SOURCE_CONCEPT_NAME, SOURCE_CONCEPT_CODE, SOURCE_VOCABULARY, CONDITION_STATUS_SOURCE_VALUE,
+      CONDITION_STATUS_CONCEPT_ID, CONDITION_STATUS_CONCEPT_NAME)
+  select a.PERSON_ID, CONDITION_CONCEPT_ID, c1.concept_name as STANDARD_CONCEPT_NAME, c1.concept_code as STANDARD_CONCEPT_CODE,
+      c1.vocabulary_id as STANDARD_VOCABULARY, CONDITION_START_DATETIME, CONDITION_END_DATETIME, CONDITION_TYPE_CONCEPT_ID,
+      c2.concept_name as CONDITION_TYPE_CONCEPT_NAME, STOP_REASON, a.VISIT_OCCURRENCE_ID, c3.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME,
+      vd.VISIT_DETAIL_ID, CONDITION_SOURCE_VALUE, CONDITION_SOURCE_CONCEPT_ID, c4.concept_name as SOURCE_CONCEPT_NAME, c4.concept_code as SOURCE_CONCEPT_CODE,
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:31]: VISIT_DETAIL_ID is  column on condition_occurrence table - no need to join visit_detail
```suggestion
      a.VISIT_DETAIL_ID, CONDITION_SOURCE_VALUE, CONDITION_SOURCE_CONCEPT_ID, c4.concept_name as SOURCE_CONCEPT_NAME, c4.concept_code as SOURCE_CONCEPT_CODE,
```

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 2]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+      c4.vocabulary_id as SOURCE_VOCABULARY, CONDITION_STATUS_SOURCE_VALUE, CONDITION_STATUS_CONCEPT_ID,
+      c5.concept_name as CONDITION_STATUS_CONCEPT_NAME
+  from \`$BQ_PROJECT.$BQ_DATASET.condition_occurrence\` a
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c1 on a.CONDITION_CONCEPT_ID = c1.CONCEPT_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c2 on a.CONDITION_TYPE_CONCEPT_ID = c2.CONCEPT_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v on a.VISIT_OCCURRENCE_ID = v.VISIT_OCCURRENCE_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c3 on v.visit_concept_id = c3.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c4 on a.CONDITION_SOURCE_CONCEPT_ID = c4.CONCEPT_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c5 on a.CONDITION_STATUS_CONCEPT_ID = c5.CONCEPT_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.visit_detail\` vd on a.VISIT_OCCURRENCE_ID = vd.VISIT_OCCURRENCE_ID"
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:41]: VISIT_DETAIL_ID is  column on condition_occurrence table - this is an unnecessary left join - please remove

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 3]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+      (PERSON_ID, DRUG_CONCEPT_ID, STANDARD_CONCEPT_NAME, STANDARD_CONCEPT_CODE, STANDARD_VOCABULARY, DRUG_EXPOSURE_START_DATETIME,
+      DRUG_EXPOSURE_END_DATETIME, VERBATIM_END_DATE, DRUG_TYPE_CONCEPT_ID, DRUG_TYPE_CONCEPT_NAME, STOP_REASON, REFILLS, QUANTITY,
+      DAYS_SUPPLY, SIG, ROUTE_CONCEPT_ID, ROUTE_CONCEPT_NAME, LOT_NUMBER, VISIT_OCCURRENCE_ID, VISIT_OCCURRENCE_CONCEPT_NAME, VISIT_DETAIL_ID, DRUG_SOURCE_VALUE,
+      DRUG_SOURCE_CONCEPT_ID, SOURCE_CONCEPT_NAME, SOURCE_CONCEPT_CODE, SOURCE_VOCABULARY, ROUTE_SOURCE_VALUE, DOSE_UNIT_SOURCE_VALUE)
+  SELECT
+      a.PERSON_ID, DRUG_CONCEPT_ID, c1.concept_name as STANDARD_CONCEPT_NAME, c1.concept_code as CONCEPT_CODE,
+      c1.vocabulary_id as STANDARD_VOCABULARY, DRUG_EXPOSURE_START_DATETIME, DRUG_EXPOSURE_END_DATETIME, VERBATIM_END_DATE,
+      DRUG_TYPE_CONCEPT_ID, c2.concept_name as DRUG_TYPE_CONCEPT_NAME, STOP_REASON, REFILLS, QUANTITY, DAYS_SUPPLY, SIG,
+      ROUTE_CONCEPT_ID, c3.concept_name as ROUTE_CONCEPT_NAME, LOT_NUMBER, a.VISIT_OCCURRENCE_ID,
+      c4.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME, vd.VISIT_DETAIL_ID, DRUG_SOURCE_VALUE, DRUG_SOURCE_CONCEPT_ID,
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:80]: VISIT_DETAIL_ID is column on condition_occurrence table - no need to join visit_detail
```suggestion
      c4.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME, a.VISIT_DETAIL_ID, DRUG_SOURCE_VALUE, DRUG_SOURCE_CONCEPT_ID,
```

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 4]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+      c5.concept_name as SOURCE_CONCEPT_NAME, c5.concept_code as SOURCE_CONCEPT_CODE, c5.vocabulary_id as SOURCE_VOCABULARY,
+      ROUTE_SOURCE_VALUE, DOSE_UNIT_SOURCE_VALUE
+  FROM \`$BQ_PROJECT.$BQ_DATASET.drug_exposure\` a
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c1 on a.DRUG_CONCEPT_ID = c1.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c2 on a.drug_type_concept_id = c2.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c3 on a.ROUTE_CONCEPT_ID = c3.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v on a.VISIT_OCCURRENCE_ID = v.VISIT_OCCURRENCE_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c4 on v.VISIT_CONCEPT_ID = c4.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c5 on a.DRUG_SOURCE_CONCEPT_ID = c5.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.visit_detail\` vd on a.VISIT_OCCURRENCE_ID = vd.VISIT_OCCURRENCE_ID"
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:90]: Please remove left join to visit_detail

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 5]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+      RANGE_LOW, RANGE_HIGH, VISIT_OCCURRENCE_ID, VISIT_OCCURRENCE_CONCEPT_NAME, VISIT_DETAIL_ID, MEASUREMENT_SOURCE_VALUE,
+      MEASUREMENT_SOURCE_CONCEPT_ID, SOURCE_CONCEPT_NAME, SOURCE_CONCEPT_CODE, SOURCE_VOCABULARY, UNIT_SOURCE_VALUE,
+      VALUE_SOURCE_VALUE)
+  select
+      a.PERSON_ID, MEASUREMENT_CONCEPT_ID, c1.concept_name as STANDARD_CONCEPT_NAME, c1.concept_code as STANDARD_CONCEPT_CODE,
+      c1.vocabulary_id as STANDARD_VOCABULARY, MEASUREMENT_DATETIME, MEASUREMENT_TYPE_CONCEPT_ID,
+      c2.concept_name as MEASUREMENT_TYPE_CONCEPT_NAME, OPERATOR_CONCEPT_ID, c3.concept_name as OPERATOR_CONCEPT_NAME,
+      VALUE_AS_NUMBER, VALUE_AS_CONCEPT_ID, c4.concept_name as VALUE_AS_CONCEPT_NAME, UNIT_CONCEPT_ID,
+      c5.concept_name as UNIT_CONCEPT_NAME, RANGE_LOW, RANGE_HIGH, a.VISIT_OCCURRENCE_ID,
+      c6.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME, vd.VISIT_DETAIL_ID, MEASUREMENT_SOURCE_VALUE, MEASUREMENT_SOURCE_CONCEPT_ID,
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:133]: VISIT_DETAIL_ID is column on condition_occurrence table - no need to join visit_detail
```suggestion
      c6.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME, a.VISIT_DETAIL_ID, MEASUREMENT_SOURCE_VALUE, MEASUREMENT_SOURCE_CONCEPT_ID,
```

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 6]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+  from \`$BQ_PROJECT.$BQ_DATASET.measurement\` a
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c1 on a.measurement_concept_id = c1.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c2 on a.measurement_type_concept_id = c2.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c3 on a.operator_concept_id = c3.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c4 on a.value_as_concept_id = c4.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c5 on a.unit_concept_id = c5.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v on a.visit_occurrence_id = v.visit_occurrence_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c6 on v.visit_concept_id = c6.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c7 on a.measurement_source_concept_id = c7.concept_id
+  left join \`$BQ_PROJECT.$BQ_DATASET.visit_detail\` vd on a.VISIT_OCCURRENCE_ID = vd.VISIT_OCCURRENCE_ID"
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:145]: Please remove left join to visit_detail

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 7]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+      UNIT_CONCEPT_NAME, VISIT_OCCURRENCE_ID, VISIT_OCCURRENCE_CONCEPT_NAME, VISIT_DETAIL_ID, OBSERVATION_SOURCE_VALUE,
+      OBSERVATION_SOURCE_CONCEPT_ID, SOURCE_CONCEPT_NAME, SOURCE_CONCEPT_CODE, SOURCE_VOCABULARY, UNIT_SOURCE_VALUE,
+      QUALIFIER_SOURCE_VALUE, value_source_concept_id, value_source_value, questionnaire_response_id)
+  SELECT
+      a.PERSON_ID, OBSERVATION_CONCEPT_ID, c1.concept_name as STANDARD_CONCEPT_NAME, c1.concept_code as STANDARD_CONCEPT_CODE,
+      c1.vocabulary_id as STANDARD_VOCABULARY, OBSERVATION_DATETIME, OBSERVATION_TYPE_CONCEPT_ID,
+      c2.concept_name as OBSERVATION_TYPE_CONCEPT_NAME, VALUE_AS_NUMBER, VALUE_AS_STRING, VALUE_AS_CONCEPT_ID,
+      c3.concept_name as VALUE_AS_CONCEPT_NAME, QUALIFIER_CONCEPT_ID, c4.concept_name as QUALIFIER_CONCEPT_NAME,
+      UNIT_CONCEPT_ID, c5.concept_name as UNIT_CONCEPT_NAME, a.VISIT_OCCURRENCE_ID, c6.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME,
+      vd.VISIT_DETAIL_ID, OBSERVATION_SOURCE_VALUE, OBSERVATION_SOURCE_CONCEPT_ID, c7.concept_name as SOURCE_CONCEPT_NAME,
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:193]: VISIT_DETAIL_ID is column on condition_occurrence table - no need to join visit_detail
```suggestion
      a.VISIT_DETAIL_ID, OBSERVATION_SOURCE_VALUE, OBSERVATION_SOURCE_CONCEPT_ID, c7.concept_name as SOURCE_CONCEPT_NAME,
```

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 8]
```diff
@@ -8,110 +8,230 @@ export BQ_PROJECT=$1  # project
...
+  FROM \`$BQ_PROJECT.$BQ_DATASET.observation\` a
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c1 on a.OBSERVATION_CONCEPT_ID = c1.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c2 on a.OBSERVATION_TYPE_CONCEPT_ID = c2.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c3 on a.value_as_concept_id = c3.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c4 on a.qualifier_concept_id = c4.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c5 on a.unit_concept_id = c5.CONCEPT_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v on a.VISIT_OCCURRENCE_ID = v.VISIT_OCCURRENCE_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c6 on v.visit_concept_id = c6.concept_id
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c7 on a.OBSERVATION_SOURCE_CONCEPT_ID = c7.CONCEPT_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.visit_detail\` vd on a.VISIT_OCCURRENCE_ID = vd.VISIT_OCCURRENCE_ID"
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:205]: Please remove left join to visit_detail

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 9]
```diff
@@ -126,27 +246,53 @@ LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c2 on a.race_concept_id = c2.CONCE
...
+  "INSERT INTO \`$BQ_PROJECT.$BQ_DATASET.ds_procedure_occurrence\`
+      (PERSON_ID, PROCEDURE_CONCEPT_ID, STANDARD_CONCEPT_NAME, STANDARD_CONCEPT_CODE, STANDARD_VOCABULARY,
+      PROCEDURE_DATETIME, PROCEDURE_TYPE_CONCEPT_ID, PROCEDURE_TYPE_CONCEPT_NAME, MODIFIER_CONCEPT_ID, MODIFIER_CONCEPT_NAME,
+      QUANTITY, VISIT_OCCURRENCE_ID, VISIT_OCCURRENCE_CONCEPT_NAME, VISIT_DETAIL_ID, PROCEDURE_SOURCE_VALUE, PROCEDURE_SOURCE_CONCEPT_ID,
+      SOURCE_CONCEPT_NAME, SOURCE_CONCEPT_CODE, SOURCE_VOCABULARY, MODIFIER_SOURCE_VALUE)
+  select
+      a.PERSON_ID, PROCEDURE_CONCEPT_ID, c1.concept_name as STANDARD_CONCEPT_NAME, c1.concept_code as STANDARD_CONCEPT_CODE,
+      c1.vocabulary_id as STANDARD_VOCABULARY, PROCEDURE_DATETIME, PROCEDURE_TYPE_CONCEPT_ID,
+      c2.concept_name as PROCEDURE_TYPE_CONCEPT_NAME, MODIFIER_CONCEPT_ID, c3.concept_name as MODIFIER_CONCEPT_NAME,
+      QUANTITY, a.VISIT_OCCURRENCE_ID, c4.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME, vd.VISIT_DETAIL_ID, PROCEDURE_SOURCE_VALUE,
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:262]: VISIT_DETAIL_ID is column on condition_occurrence table - no need to join visit_detail
```suggestion
      QUANTITY, a.VISIT_OCCURRENCE_ID, c4.concept_name as VISIT_OCCURRENCE_CONCEPT_NAME, a.VISIT_DETAIL_ID, PROCEDURE_SOURCE_VALUE,
```

File: api/db-cdr/generate-cdr/build-ds-tables.sh [hunk 10]
```diff
@@ -126,27 +246,53 @@ LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c2 on a.race_concept_id = c2.CONCE
...
+      PROCEDURE_SOURCE_CONCEPT_ID, c5.concept_name as SOURCE_CONCEPT_NAME, c5.concept_code as SOURCE_CONCEPT_CODE,
+      c5.vocabulary_id as SOURCE_VOCABULARY, MODIFIER_SOURCE_VALUE
+  from \`$BQ_PROJECT.$BQ_DATASET.procedure_occurrence\` a
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c1 on a.PROCEDURE_CONCEPT_ID = c1.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c2 on a.PROCEDURE_TYPE_CONCEPT_ID = c2.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c3 on a.MODIFIER_CONCEPT_ID = c3.CONCEPT_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.visit_occurrence\` v on a.VISIT_OCCURRENCE_ID = v.VISIT_OCCURRENCE_ID
+  left join \`$BQ_PROJECT.$BQ_DATASET.concept\` c4 on v.visit_concept_id = c4.concept_id
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.concept\` c5 on a.PROCEDURE_SOURCE_CONCEPT_ID = c5.CONCEPT_ID
+  LEFT JOIN \`$BQ_PROJECT.$BQ_DATASET.visit_detail\` vd on a.VISIT_OCCURRENCE_ID = vd.VISIT_OCCURRENCE_ID"
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-ds-tables.sh:272]: Please remove left join to visit_detail

**Review by freemabd — COMMENTED** [review]
I forgot to add changes for the creation of the tables: https://github.com/all-of-us/workbench/blob/5a279f27ba74a4c1d35c0ea97bb4daad49200177/api/db-cdr/generate-cdr/create-tables.sh#L21-L21
Basically you need to add a new json file for ds_procedure_occurrence_v52.json that doesn't have visit_detail_id and rename of qualifier_source_value. This script needs to check if visit_detail exists and only load ds_procedure_occurrence.json or ds_procedure_occurrence_v52.json

File: api/db-cdr/generate-cdr/build-ds-tables.sh (hunk 7 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-ds-tables.sh]: removed left join and referenced the to a.VISIT_DETAIL_ID

File: api/db-cdr/generate-cdr/build-ds-tables.sh (hunk 6 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-ds-tables.sh]: done and referenced the to a.VISIT_DETAIL_ID

File: api/db-cdr/generate-cdr/build-ds-tables.sh (hunk 5 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-ds-tables.sh]: removed left join and referenced the to a.VISIT_DETAIL_ID

File: api/db-cdr/generate-cdr/build-ds-tables.sh (hunk 4 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-ds-tables.sh]: done and referenced the to a.VISIT_DETAIL_ID

File: api/db-cdr/generate-cdr/build-ds-tables.sh (hunk 3 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-ds-tables.sh]: done and referenced the to a.VISIT_DETAIL_ID

File: api/db-cdr/generate-cdr/build-ds-tables.sh (hunk 2 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-ds-tables.sh]: done and referenced the to a.VISIT_DETAIL_ID

File: api/db-cdr/generate-cdr/build-ds-tables.sh (hunk 1 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-ds-tables.sh]: done and referenced the to a.VISIT_DETAIL_ID

File: api/db-cdr/generate-cdr/bq-schemas/ds_condition_occurrence_52.json [hunk 11]
```diff
@@ -0,0 +1,82 @@
+[
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/bq-schemas/ds_condition_occurrence_52.json:1]: This file is not needed. Please remove

File: api/db-cdr/generate-cdr/bq-schemas/ds_drug_exposure_52.json [hunk 12]
```diff
@@ -0,0 +1,110 @@
+[
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/bq-schemas/ds_drug_exposure_52.json:1]: This file is not needed. Please remove

File: api/db-cdr/generate-cdr/bq-schemas/ds_measurement_52.json [hunk 13]
```diff
@@ -0,0 +1,106 @@
+[
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/bq-schemas/ds_measurement_52.json:1]: This file is not needed. Please remove

File: api/db-cdr/generate-cdr/bq-schemas/ds_observation_52.json [hunk 14]
```diff
@@ -0,0 +1,114 @@
+[
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/bq-schemas/ds_observation_52.json:1]: This file is not needed. Please remove

File: api/db-cdr/generate-cdr/build-backup-cb-ds-tables.sh [hunk 15]
```diff
@@ -54,8 +54,13 @@ ds_zip_code_socioeconomic
 
 for t in "${backup_tables[@]}"
 do
+    t_schema=$t".json"
+    if [[ ! "$TABLE_LIST" == *"visit_detail"* &&  "ds_condition_occurrence|ds_drug_exposure|ds_measurement|ds_observation|ds_procedure_occurrence" =~ $t ]]
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/build-backup-cb-ds-tables.sh:58]:
```suggestion
    if [[ ! "$TABLE_LIST" == *"visit_detail"* &&  "ds_procedure_occurrence" =~ $t ]]
```

File: api/db-cdr/generate-cdr/create-tables.sh [hunk 16]
```diff
@@ -37,6 +37,46 @@ do
...
+      bq --quiet --project_id="$BQ_PROJECT" mk --schema="$schema_path/$json_name" --time_partitioning_type=DAY --clustering_fields person_id,domain "$BQ_DATASET.$table_name"
+    elif [[ "$table_name" == 'ds_observation_52' && "$TABLE_LIST" =~ *'visit_detail'* ]]
+    then
+      echo "Creating $table_name"
+      bq --quiet --project_id="$BQ_PROJECT" mk --schema="$schema_path/$json_name" --time_partitioning_type=DAY --clustering_fields person_id,domain "$BQ_DATASET.$table_name"
+    elif [[ "$table_name" == 'ds_procedure_occurrence' && "$TABLE_LIST" == *'visit_detail'* ]]
+    then
+      echo "Creating $table_name"
+      bq --quiet --project_id="$BQ_PROJECT" mk --schema="$schema_path/$json_name" --time_partitioning_type=DAY --clustering_fields person_id,domain "$BQ_DATASET.$table_name"
+    elif [[ "$table_name" == 'ds_procedure_occurrence_52' && "$TABLE_LIST" =~ *'visit_detail'* ]]
```
Comment (freemabd) [inline api/db-cdr/generate-cdr/create-tables.sh:79]: Please remove these conditions. We only need to add a condition for ds_procedure_occurrence and it should only be created in the default else -> must be unclustered.

**Review by freemabd — COMMENTED** [review]
Have a few problems to address.

File: api/db-cdr/generate-cdr/create-tables.sh (hunk 16 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/create-tables.sh:79]: I did not follow what this means, but ok. removed all the condition-lines above

File: api/db-cdr/generate-cdr/bq-schemas/ds_observation_52.json (hunk 14 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/bq-schemas/ds_observation_52.json:1]: removed

File: api/db-cdr/generate-cdr/bq-schemas/ds_measurement_52.json (hunk 13 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/bq-schemas/ds_measurement_52.json:1]: removed

File: api/db-cdr/generate-cdr/bq-schemas/ds_drug_exposure_52.json (hunk 12 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/bq-schemas/ds_drug_exposure_52.json:1]: removed

File: api/db-cdr/generate-cdr/bq-schemas/ds_condition_occurrence_52.json (hunk 11 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/bq-schemas/ds_condition_occurrence_52.json:1]: removed

File: api/db-cdr/generate-cdr/build-backup-cb-ds-tables.sh (hunk 15 above)
Comment (chenchals) [reply api/db-cdr/generate-cdr/build-backup-cb-ds-tables.sh:58]: changed

**Review by freemabd — APPROVED** [review]
Looks good.

**Pull request merged**